### PR TITLE
Updating more custom base uri test to include missing param and tests

### PR DESCRIPTION
**Pull request by catalinaperalta**
Adding more custom uri test and modified/updated the corresponding generated code to add missing params.

### CODE REVIEW

File: test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go [hunk 1]
```diff
@@ -6,18 +6,21 @@
...
 )
 
 type PathsOperations struct{}
 
 // GetEmptyCreateRequest creates the GetEmpty request.
-func (PathsOperations) GetEmptyCreateRequest(u url.URL, vault string, secret string, keyName string, keyVersion string) (*azcore.Request, error) {
-	u.Path = path.Join(u.Path, "/customuri/{subscriptionId}/{keyName}")
-	return azcore.NewRequest(http.MethodGet, u), nil
+func (PathsOperations) GetEmptyCreateRequest(u url.URL, vault string, secret string, keyName string, keyVersion string, subscriptionID string) (*azcore.Request, error) {
+	u.Path = path.Join(u.Path, "/customuri", subscriptionID, keyName)
```
Comment (jhendrixMSFT) [inline test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go]: I like this solution.  Is there a bug tracking fixing this in the generator?

Comment (catalinaperalta) [reply test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go]: I added it as part of my general list of bugs that I've found while doing the tests but I'll also add it's own specific one. 

Also for some reason both the DNS and subscriptionID param were not added. I wonder if it has to do with how they're referenced in the swagger.

Comment (catalinaperalta) [reply test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go]: Here's the issue: #263

Comment (jhendrixMSFT) [reply test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go]: For the missing parameters, my guess is they're defined as `client` which in track 1 meant they were parameters on the client constructor.  For the internal package we'll need to add them in the regular list of params.  And for the convenience layer we'll likely need to add them as constructor params.

Comment (catalinaperalta) [reply test/autorest/generated/morecustombaseurigroup/internal/morecustombaseurigroup/paths.go]: Ah probably. I'm going to add your last two sentences as comments in the issue.

**Pull request merged**